### ISSUE TITLE
Question about the selection strategy of the final prompt

### ISSUE THREAD

**Issue by jihyukkim-nlp** [XinyuanWangCS/PromptAgent#11]
Hello, really appreciate for publishing this interesting paper and codes! 

I have a question about the selection strategy of the final prompt.
In the README file (and also in the paper), it said "the highest reward node in the highest average reward path" is used for the final  prompt. 
I am wondering why the highest reward node should be selected "from the highest average reward path"? 
I thought selecting the highest reward node "from the entire tree" seems more intuitive.
Is considering the average reward on each path to prevent the prompt from overfitting?

**Comment by XinyuanWangCS** [comment]
Hi,

There are several possible selection strategies (best global reward, best path's best reward, best Q...) which align our intuitions. "prevent the prompt from overfitting" is exactly why we use the current strategy. What PromptAgent does is like fitting the tree towards higher rewards calculated on the valication set. The validation set is normal a small portion of the whole dataset. It could be IID sampled, but still we can't guarantee the highest validation reward will lead to the highest test metric, most of the time it doesn't. We find sometimes the test metric of the prompt with highest reward is lower than the one with a bit lower reward. We think it's due to overfitting issue as you said. Picking the best path first can add some robustness and make the result more stable.

**Comment by jihyukkim-nlp** [comment]
Thank you for your prompt reply and the clarification!
